choreolib/src/main/java/choreo/Choreo.java

```diff
@@ -431,4 +434,53 @@ public static <SampleType extends TrajectorySample<SampleType>> AutoFactory crea
         Optional.of(trajectoryLogger),
         requireNonNullParam(alliance, "alliance", "Choreo.createAutoFactory"));
   }
+
+  /**
+   * Creates an alert under the "Choreo" group.
+   *
+   * @param name The name of the alert
+   * @param type The type of alert
+   * @return an Alert published under the "Choreo" group
+   */
+  public static Alert alert(String name, Alert.AlertType type) {
+    return new Alert("Choreo", name, type);
+  }
+
+  /**
+   * Creates a {@link MultiAlert} under the "Choreo" group.
+   *
+   * @param textGenerator A function that accepts a list of causes and returns an alert message
+   * @param type The type of alert
+   * @return a MultiAlert published under the "Choreo" group
+   */
+  public static MultiAlert multiAlert(
+      Function<List<String>, String> textGenerator, Alert.AlertType type) {
+    return new MultiAlert(textGenerator, type);
+  }
+
+  /**
+   * An alert that can have multiple causes. Utilizes a function to generate an error message from a
+   * list of causes.
+   */
+  public static class MultiAlert extends Alert {
```

Comment:
This seems like it could be a common usecase do we wanna upstream it?

Reply:
Sure I can PR it to wpilib later; the pr process might take some time though(we can change to the PR'd version in a future pull request)